### PR TITLE
API 800, 1000, 1200 support for Server Profile and Server Profile Template

### PR DESCRIPTION
**Pull request by jyothisgm**
### Description
API 800, 1000, 1200 support for Server Profile and Server Profile Template

### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

**Review by AnushaY1916 — COMMENTED** [review]
Update the license in the example file which you modified.

File: libraries/resource_providers/api1200/synergy/server_profile_template_provider.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API1200
+    module Synergy
+      # Server Profile Template API1200 Synergy provider
+      class ServerProfileTemplateProvider < OneviewCookbook::API1000::C7000::ServerProfileTemplateProvider
```
Comment (soodpr) [inline libraries/resource_providers/api1200/synergy/server_profile_template_provider.rb]: i think it should inherit from API1200 keeping it sync with other API's implementation. Though in this case, there won't be any failure but in future, if we are adding in 1200 API, it should get inherit in Synergy API 1200 as well.

Comment (jyothisgm) [reply libraries/resource_providers/api1200/synergy/server_profile_template_provider.rb]: Fixed

**Comment by jyothisgm** [comment]
> Update the license in the example file which you modified.

Done

**Review by soodpr — APPROVED** [review]
LGTM 👍

**Review by AnushaY1916 — APPROVED** [review]

**Pull request merged**